pb-rs/src/parser.rs

```diff
@@ -400,29 +430,54 @@ fn enum_field(input: &str) -> IResult<&str, (String, i32)> {
     )(input)
 }
 
+fn enum_event(input: &str) -> IResult<&str, EnumEvent> {
+    alt((
+        map(enum_field, EnumEvent::Field),
+        value(EnumEvent::Ignore, option_ignore),
+        value(EnumEvent::Ignore, br),
+    ))(input)
+}
+
 fn enumerator(input: &str) -> IResult<&str, Enumerator> {
     map(
         terminated(
             pair(
                 delimited(pair(tag("enum"), many1(br)), word, many0(br)),
-                delimited(
-                    pair(tag("{"), many0(br)),
-                    separated_list0(many0(br), enum_field),
-                    pair(many0(br), tag("}")),
-                ),
+                delimited(tag("{"), many0(enum_event), tag("}")),
             ),
             opt(pair(many0(br), tag(";"))),
         ),
-        |(name, fields)| Enumerator {
+        |(name, events)| {
+            let mut enumerator = Enumerator {
             name,
-            fields,
             ..Default::default()
+        };
+        for event in events {
+            if let EnumEvent::Field(f) = event {
+                enumerator.fields.push(f);
+            }
+        }
+        enumerator
         },
     )(input)
 }
 
 fn option_ignore(input: &str) -> IResult<&str, ()> {
-    value((), delimited(tag("option"), take_until(";"), tag(";")))(input)
+    value((), delimited(pair(tag("option"), many1(br)), take_until(";"), tag(";")))(input)
```

Comment:
I changed this rule to ensure that this parser only matches `option ...` and not `optional ...`

Reply:
Nice catch!